ci(integration-tests): Add workflow for core integration tests. (#2334)
y-scope/clp: docs/src/dev-docs/tooling-gh-workflows.md

This workflow is responsible for:

1. building (Linux) container images containing CLP-core's dependencies,

> 2. building CLP-core and running its unit & integration tests, and 

Reviewer: ```suggestion
2. building CLP-core and running its unit & integration tests, and
```

Author: agreed

  core's unit tests.
* `ubuntu-jammy-binaries-image`: Builds an Ubuntu Jammy container image containing CLP-core's
  binaries built in the `ubuntu-jammy-binaries` job.
* `ubuntu-jammy-integration-tests-core`: Runs CLP-core's integration tests using the binaries built

>   in the `ubuntu-jammy-binaries` job. The test logs are uploaded at the end of the test run.

Reviewer: ```suggestion
  in the `ubuntu-jammy-binaries` job, and then uploads the logs from the tests.
```

Author: agreed
